runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -810,10 +810,21 @@ UDATA *
 TR_J9JITaaSServerSharedCache::rememberClass(J9Class *clazz, bool create)
    {
    TR_ASSERT(_stream, "stream must be initialized by now");
-   // TODO: might want to keep a per-client cache of (class -> class chain)
-   // that were remembered to reduce the number of remote calls
+   PersistentUnorderedMap<J9Class *, UDATA *> & cache = TR::compInfoPT->getClientData()->getClassClainDataCache();
```

Comment:
Avoid using TR::compInfoPT repeatedly. This is using thread local storage which is expensive to access, so try to use as little as possible if at all. Here we can just remember the clientDta